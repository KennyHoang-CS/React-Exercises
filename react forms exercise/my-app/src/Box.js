import React from 'react';

const Box = ({ id, color, width, height, deleteBox }) => {
    return (
        <div>
            <div className="Box"
                id={id}
                style={{
                    backgroundColor: color,
                    width: parseInt(width) || 100,
                    height: parseInt(height) || 100
            }}
            >  
            <button onClick={() => deleteBox(id)} data-testid="delete-box">X</button>
            </div>
        </div>
    )
}

export default Box;