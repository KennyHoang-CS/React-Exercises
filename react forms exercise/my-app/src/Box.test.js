import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Box from './Box';


it('Box should render', () => {
    render(<Box />);
});

it('Box render matches snapshot', () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot(); 
});
