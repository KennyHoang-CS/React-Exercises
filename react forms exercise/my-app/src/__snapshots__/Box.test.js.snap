// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box render matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Box"
      style="width: 100px; height: 100px;"
    >
      <button
        data-testid="delete-box"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
